src/dune_engine/sandbox.ml

```diff
@@ -245,7 +233,10 @@ let move_targets_to_build_dir t ~loc ~should_be_skipped
         rename_optional_file ~src:(map_path t target) ~dst:target);
   let discovered_targets =
     Path.Build.Set.to_list_map targets.dirs ~f:(fun target ->
-        rename_dir_recursively ~loc ~src_dir:(map_path t target) ~dst_dir:target)
+        let src_dir = map_path t target in
+        let files = collect_dir_recursively ~loc ~src_dir ~dst_dir:target in
+        Path.rename (Path.build src_dir) (Path.build target);
```

Comment:
Maybe copy the above code error here in case the rename fails?

Reply:
I don't think it can fail because dst exists anymore. Since rename will simply overwrite dst in such a case.

Reply:
Hmm, good point. But shouldn't we report an error instead of silently overwriting what's there?

Reply:
I could add a check, but it doesn't seem to be worth it tbh. We already manually clear the stale targets, and this call would nuke them once more. Seems a bit gratuitous.

Reply:
> We already manually clear the stale targets

That's the current behaviour/invariant but we might accidentally break it in future, and in that case, it would be nice if tests catch us. And they can't if we don't fail here.

I don't have a very strong opinion here. I'm just pointing out that previously we would fail here when coming across a stale directory, and now we don't, which feels like a step in the wrong direction.

Reply:
Okay, I can add it back. It's one extra syscall, and we've already shaved off like a 100 in this code path :)

Reply:
How about having a "paranoid" mode used in tests only, so we don't need to pay the extra cost all the time?

Reply:
Okay, I can add the the check in this PR. But then add guard it another PR.

Reply:
Thanks, sounds good!